src/Std/Data/DTreeMap/Lemmas.lean

```diff
@@ -3229,6 +3229,140 @@ theorem minKey!_alter_eq_self [TransCmp cmp] [Inhabited α] {k f}
 
 end Const
 
+theorem minKey?_eq_some_minKeyD [TransCmp cmp] (he : t.isEmpty = false) {fallback} :
+    t.minKey? = some (t.minKeyD fallback) :=
+  Impl.minKey?_eq_some_minKeyD t.wf he
+
```

Comment:
I haven't been able to find the lemma `minKey_eq_minKey!` (or should it be stated in the other direction?), but it's possible that it's there in one of the PRs and I just haven't seen it.

Reply:
I looked them up and ypu are right. I decided to add them later in #7669 together with `maxKey!`.